### ISSUE TITLE
Apache Superset is not starting line 50: /usr/bin/run-server.sh: No such file or directory

### ISSUE THREAD

**Issue by ninomllr** [apache/superset#17712]
Apache Superset docker-compose is not working for a newly cloned project and by following the offical documentation: https://superset.apache.org/docs/installation/installing-superset-using-docker-compose

#### How to reproduce the bug

1. Clone the superset git locally
2. Run `docker-compose -f docker-compose-non-dev.yml up`

### Expected results

The application should start.

### Actual results

The web application is not starting and the log shows the following error:
```
superset_app            | Starting web app...
superset_app            | /app/docker/docker-bootstrap.sh: line 50: /usr/bin/run-server.sh: No such file or directory
```

### Environment

git checkout today 10.12.2021, no changes on my side just the latest repository

**Comment by atpage** [comment]
Same here, probably due to this change?  https://github.com/apache/superset/commit/aee5c9a3ea1006b194319d47c3e7f245e438ce6f

**Comment by atpage** [comment]
I confirmed that it works as of the previous commit (8c25f2f356e488120d7ce3ff852a4b6b9a049b3b).

**Comment by ninomllr** [comment]
That was it! When I pulled the images again, it worked.

**Comment by ctmay4** [comment]
I just checked out master and tried to run through Docker and I am getting the same error.  If I check out the `1.4.0` tag it starts correctly without errors.  According to the documentation `master` is supposed to work, right?

```
superset_app            | Starting web app...
superset_app            | Skipping local overrides
superset_app            | Starting web app...
superset_app            | /app/docker/docker-bootstrap.sh: line 50: /usr/bin/run-server.sh: No such file or directory
superset_app            | Skipping local overrides
superset_app            | Starting web app...
superset_app            | /app/docker/docker-bootstrap.sh: line 50: /usr/bin/run-server.sh: No such file or directory
superset_app            | Skipping local overrides
superset_app            | Starting web app...
superset_app            | /app/docker/docker-bootstrap.sh: line 50: /usr/bin/run-server.sh: No such file or directory
superset_app            | Skipping local overrides
superset_app            | Starting web app...
superset_app            | /app/docker/docker-bootstrap.sh: line 50: /usr/bin/run-server.sh: No such file or directory
superset_app            | Skipping local overrides
```

**Comment by atpage** [comment]
Still broken for me as well.  @amitmiran137 any reason this was closed?

**Comment by amitmiran137** [comment]
@ofekisr can you check it out ?

**Comment by wiktor2200** [comment]
The problem exists in 1.4.0. There shouldn't be so many differences and long periods of time between releases, because there are a lot of changes everywhere and there is no way to use default helm chart values with stable version. :(
Yesterday release is based on commit from OCTOBER 2021 which causes 3,5months of fixes and improvements missing in new release.
There is 285 changed files with 8,931 additions and 22,463 deletions difference between current `master` and release tag `1.4.0`
@eschutho is there any roadmap for release 1.4.1 or any timetable when next release will cover those changes. 

It would be good to add some tests for helm chart deployment for both `latest` and stable version.

**Comment by ofekisr** [comment]
first, why use docker-compose when you can use the helm chart?
second, just need to change the docker-compose bootstrap file to the new one.
@ninomllr @amitmiran137 @atpage

**Comment by atpage** [comment]
@ofekisr We are following these instructions:
https://superset.apache.org/docs/installation/installing-superset-using-docker-compose

**Comment by ofekisr** [comment]
@atpage @amitmiran137 
anyway IMO it is not my change, if you look inside the Dockerfile you'll see the bootstrap script file is copied only inside the CI version while the docker-compose file reference the official release image

**Comment by rafalpas** [comment]
@wiktor2200 I have found this discussion for 1.4.0 bugs that are candidates for being fixed in 1.4.1 -> https://github.com/apache/superset/discussions/18119

**Comment by nytai** [comment]
Running 
`docker-compose -f docker-compose-non-dev.yml pull` before running the up command should fix this. The issue is that the version of code you are on is not compatible with the docker image you have, so you need to pull a newer docker image that is compatible with this code

**Comment by nytai** [comment]
Running the pull command should be standard practice whenever you pull the latest source code

**Comment by eschutho** [comment]
> The problem exists in 1.4.0. There shouldn't be so many differences and long periods of time between releases, because there are a lot of changes everywhere and there is no way to use default helm chart values with stable version. :( Yesterday release is based on commit from OCTOBER 2021 which causes 3,5months of fixes and improvements missing in new release. There is 285 changed files with 8,931 additions and 22,463 deletions difference between current `master` and release tag `1.4.0` @eschutho is there any roadmap for release 1.4.1 or any timetable when next release will cover those changes.
> 
> It would be good to add some tests for helm chart deployment for both `latest` and stable version.

@wiktor2200 1.4.1 will only include bug fixes (not features) that address issues on 1.4.0, so of those 285 changes, only a small portion will apply. 1.4.1 is scheduled to go for vote in the next few days. Please add any bug fixes to the discussion here: https://github.com/apache/superset/discussions/18119

**Comment by manisha-tanwar** [comment]
seeing same issue when deploying it using Helm and same on version 1.3.2.
I don't want to use latest as it shows version as v.0.0.0dev on UI.

**Comment by gai6948** [comment]
`1.4.1` still has the same issue for me

**Comment by eschutho** [comment]
@manisha-tanwar @gai6948 have you tried @nytai's suggestion to pull the docker container again before running it?

**Comment by gai6948** [comment]
It was problem with the helm chart, because running in docker-compose was perfectly fine.

**Comment by arkpoah** [comment]
Same issue for me with 0.5.8 chart with 1.4.1 or 1.4.0 image

**Comment by eschutho** [comment]
Ok, that's what I expected, so for now, <= helm 0.4.0 works with all of the official Superset releases up to 1.4.1. Helm >=0.5.0 will be compatible with either the "latest" non release of Superset and the upcoming 1.5.0 release and is not compatible with <= Superset 1.4.1.

**Comment by fertek** [comment]
With `x-superset-image: &superset-image apache/superset:1.4.1-dev` in `docker-compose-non-dev.yml` running `docker-compose -f docker-compose-non-dev.yml pull` before `docker-compose -f docker-compose-non-dev.yml up` does not help:

```
superset_app            | Skipping local overrides
superset_app            | Starting web app...
superset_app            | /app/docker/docker-bootstrap.sh: line 50: /usr/bin/run-server.sh: No such file or directory
```

**Comment by caleb15** [comment]
Same issue with latest superset helm chart, tag: 1.4.2. I pulled 1.4.2 and looked inside. run-server does not exist in the docker image. @villebro can this be reopened please?

```
➜  ~ docker pull apache/superset:1.4.2
1.4.2: Pulling from apache/superset
e4d61adff207: Pull complete
4ff1945c672b: Pull complete
ff5b10aec998: Pull complete
12de8c754e45: Pull complete
ada1762e7602: Pull complete
2f2b2e030155: Pull complete
4169ad44c6d7: Pull complete
672a77ed36b6: Pull complete
a6d75b62b7e9: Pull complete
4fe7ffd93a2a: Pull complete
537246e6af6d: Pull complete
5f6fb23fb0de: Pull complete
f926f3e1899a: Pull complete
33c99eb618fb: Pull complete
6759fae3c280: Pull complete
6ad24bb852c9: Pull complete
b7b33bbfd8da: Pull complete
71cbd0b2a3f1: Pull complete
Digest: sha256:a85b0503586f7c08cfe2a9e3654ab155bb674756fff54759f892bcb83853c530
Status: Downloaded newer image for apache/superset:1.4.2
docker.io/apache/superset:1.4.2
➜  ~docker run -it apache/superset:1.4.2 -- bash
➜  ~superset@6bc1707857d5:/app$ cat /usr/bin/run-server.sh
cat: /usr/bin/run-server.sh: No such file or directory
```

**Comment by eschutho** [comment]
@caleb15 this issue should be resolved in the upcoming 1.5.0 release.

**Comment by anantmulchandani** [comment]
facing this issue in superset 4.1.1 in a dockerised setup.
```
superset_app          | Error: No application module specified.
superset_app          | /usr/bin/run-server.sh: line 37: superset.app:create_app(): command not found
superset_app exited with code 127
```
